exonum/src/events/noise/wrapper.rs

```diff
@@ -105,7 +105,7 @@ impl NoiseWrapper {
                 msg.len()
             };
 
-            let (_, read_to) = self.read(msg, len_to_read).unwrap();
+            let read_to = self.read(msg, len_to_read).unwrap();
```

Comment:
Like with writing, two suggestions:

- Explain why `unwrap()` is OK here.
- Check the length of the read message.

Also, if the length of `decoded_message` deterministically depends on the length of `data`, would it make sense to pre-allocate correspondingly?